.circleci/config.yml

```diff
@@ -68,6 +68,94 @@ jobs:
         make images-nodep
         make push-nodep
 
+  e2e:
+    machine: true
+    environment:
+      CHANGE_MINIKUBE_NONE_USER: true
+      KUBERNETES_VERSION: v1.10.0
+      MINIKUBE_VERSION: v0.27.0
+    steps:
+    - run:
+        name: Configure CI instance
+        command: |
+          # Make root rshared in order to fix kube-dns problems.
+          sudo mount --make-rshared /
+          # Create project directory
+          sudo mkdir -p /go/src/github.com/kubermatic/cluster-api-provider-digitalocean
+          # Set correct permissions on project directory
+          sudo chown -R $USER:$USER /go
+    - restore_cache:
+        key: repo-{{ .Environment.CIRCLE_SHA1 }}
+    - run:
+        name: Install dependencies
+        command: |
+          # Download minikube.
+          curl -Lo minikube https://storage.googleapis.com/minikube/releases/${MINIKUBE_VERSION}/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+          # Download kubeadm.
+          curl -Lo kubeadm https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubeadm && chmod +x kubeadm && sudo mv kubeadm /usr/bin/
+          # Download kubectl, which is a requirement for using minikube and running e2e tests.
+          curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+    - run:
+        name: Set up minikube
+        command: |
+          # Start minikube.
+          sudo minikube start --vm-driver=none --bootstrapper=localkube --kubernetes-version=${KUBERNETES_VERSION} --feature-gates=CustomResourceSubresources=true --extra-config=apiserver.Authorization.Mode=RBAC
+          # Set correct permissions on minikube and kubeconfig files.
+          sudo chown -R $USER:$USER $HOME/.kube
+          sudo chown -R $USER:$USER $HOME/.minikube
+          # Update the kubeconfig to use the minikube cluster.
+          minikube update-context
+    - run:
+        name: Update kubeconfig
+        command: |
+          # Read certificate files.
+          MINIKUBE_CA_CERT=$(cat $HOME/.minikube/ca.crt | base64 -w0)
+          MINIKUBE_CLIENT_CERT=$(cat $HOME/.minikube/client.crt | base64 -w0)
+          MINIKUBE_CLIENT_CERT_KEY=$(cat $HOME/.minikube/client.key | base64 -w0)
+          # Update kubeconfig file.
+          sed -i "s/certificate-authority.*/certificate-authority-data: $MINIKUBE_CA_CERT/" $HOME/.kube/config
```

Comment:
To be checked.

Reply:
Feature allowing to automatically embed certificates is available as of Minikube v0.29.0.

Minikube v0.28+ doesn't work correctly with clusterctl, and Minikube v0.29.0 doesn't ship with localkube bootstrap anymore, so it can't be used in CI.